File: src/demos/ps/maps/Bortevo.java
package demos.ps.maps;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import static demos.ps.oo.PSGame.EntFinish;
import static demos.ps.oo.PSGame.EntStart;
import static demos.ps.oo.PSGame.Shop;
import static demos.ps.oo.PSGame.getItem;
import static demos.ps.oo.PSGame.getString;
import static demos.ps.oo.PSGame.getYesNo;
import static demos.ps.oo.PSGame.mapswitch;
import demos.ps.oo.City;
import demos.ps.oo.Item;
import demos.ps.oo.PSGame;
import demos.ps.oo.PSGame.Flags;
import demos.ps.oo.PSGame.Planet;
import demos.ps.oo.PSLibItem.OriginalItem;
import demos.ps.oo.PSMenu;
import demos.ps.oo.PSMenu.EntityClothes;
import demos.ps.oo.PSMenu.EntityType;
import demos.ps.oo.PSMenu.LargeEntity;
import demos.ps.oo.PSMenu.Scene;
import demos.ps.oo.PSMenu.SpecialEntity;

public class Bortevo {

	private static final Logger log = LogManager.getLogger(Bortevo.class);

	
	public static void house() {
		PSMenu.startScene(Scene.RUINED_HOUSE, SpecialEntity.BEGGAR);
		PSMenu.Stext(getString("Bortevo_People_Man_2"));
		PSMenu.endScene();
	}

	public static void hapsby() {
		if(!PSGame.hasFlag(Flags.GOT_HAPSBY)) {
			PSMenu.startScene(Scene.RUINED_HOUSE, LargeEntity.JUNK);
		} else {
			PSMenu.startScene(Scene.RUINED_HOUSE, SpecialEntity.NONE);
		}
		
		PSMenu.instance.waitB1();
		if(!PSGame.hasFlag(Flags.GOT_HAPSBY) && PSGame.getParty().hasQuestItem(getItem(OriginalItem.Quest_Polymeteral))) {
			PSMenu.Stext(getString("Bortevo_House_Hapsby"));
			PSMenu.startScene(Scene.SCREEN_NOFADE, LargeEntity.HAPSBY);
			PSMenu.instance.waitDelay(30);
			PSMenu.Stext(getString("Bortevo_Hapsby_Free"));
			PSMenu.StextNext(getString("Bortevo_Hapsby"));
			PSGame.getParty().removeItem(getItem(OriginalItem.Quest_Polymeteral));
			PSGame.setFlag(Flags.GOT_HAPSBY);
			
		} else {
			PSMenu.Stext(getString("Chest_Search"));
		}
		PSMenu.endScene();
	}
	
	public static void junk() {
		PSMenu.startScene(Scene.RUINED_HOUSE, LargeEntity.JUNK);
		PSMenu.instance.waitB1();
		PSMenu.Stext(getString("Chest_Search"));
		PSMenu.endScene();
	}	
	
	public static void hovercraft() {
		PSMenu.startScene(Scene.RUINED_HOUSE, LargeEntity.JUNK);
		PSMenu.instance.waitB1();
		if(PSGame.hasFlag(Flags.INFO_HOVER) && !PSGame.getParty().hasQuestItem(getItem(OriginalItem.Vehicle_FlowMover))) {
			PSMenu.Stext(getString("Bortevo_House_Hovercraft"));
			PSGame.getParty().addQuestItem(getItem(OriginalItem.Vehicle_FlowMover));
		} else {
			PSMenu.Stext(getString("Chest_Search"));
		}
		PSMenu.endScene();
	}

	public static void ent1() {
		EntStart();
		PSMenu.Stext(getString("Bortevo_People_Man_1"));
		EntFinish();
	}

	public static void exit() {
		mapswitch(Planet.PALMA, City.BORTEVO.getX(), City.BORTEVO.getY());		
	}

	
	
}

File: src/demos/ps/maps/Bortevo.ts
/**
 * Bortevo City Script
 * TypeScript port of Bortevo.java
 */

import { PSGame } from '../PSGame';
import { Flags } from '../game/GameData';
import { Planet, City, CityHelper } from '../game/City';
import { OriginalItem } from '../game/PSLibItem';
import { PSSceneType, SpecialEntity, LargeEntity, PSMenu } from '../PSMenu';

export class Bortevo {

  public static async house(): Promise<void> {
    await PSMenu.startScene(PSSceneType.RUINED_HOUSE, SpecialEntity.BEGGAR);
    await PSMenu.Stext(PSGame.getString("Bortevo_People_Man_2"));
    await PSMenu.endScene();
  }

  public static async hapsby(): Promise<void> {
    if (!PSGame.hasFlag(Flags.GOT_HAPSBY)) {
      await PSMenu.startSceneWithLargeEntity(PSSceneType.RUINED_HOUSE, LargeEntity.JUNK);
    } else {
      await PSMenu.startScene(PSSceneType.RUINED_HOUSE, SpecialEntity.NONE);
    }

    await PSMenu.instance.waitB1();
    if (!PSGame.hasFlag(Flags.GOT_HAPSBY) && PSGame.getParty().hasQuestItem(PSGame.getItem(OriginalItem.Quest_Polymeteral))) {
      await PSMenu.Stext(PSGame.getString("Bortevo_House_Hapsby"));
      await PSMenu.startSceneWithLargeEntity(PSSceneType.SCREEN_NOFADE, LargeEntity.HAPSBY);
      await PSMenu.instance.waitDelay(30);
      await PSMenu.Stext(PSGame.getString("Bortevo_Hapsby_Free"));
      await PSMenu.StextNext(PSGame.getString("Bortevo_Hapsby"));
      PSGame.getParty().removeItem(PSGame.getItem(OriginalItem.Quest_Polymeteral));
      PSGame.setFlag(Flags.GOT_HAPSBY);
    } else {
      await PSMenu.Stext(PSGame.getString("Chest_Search"));
    }
    await PSMenu.endScene();
  }

  public static async junk(): Promise<void> {
    await PSMenu.startSceneWithLargeEntity(PSSceneType.RUINED_HOUSE, LargeEntity.JUNK);
    await PSMenu.instance.waitB1();
    await PSMenu.Stext(PSGame.getString("Chest_Search"));
    await PSMenu.endScene();
  }

  public static async hovercraft(): Promise<void> {
    await PSMenu.startSceneWithLargeEntity(PSSceneType.RUINED_HOUSE, LargeEntity.JUNK);
    await PSMenu.instance.waitB1();
    if (PSGame.hasFlag(Flags.INFO_HOVER) && !PSGame.getParty().hasQuestItem(PSGame.getItem(OriginalItem.Vehicle_FlowMover))) {
      await PSMenu.Stext(PSGame.getString("Bortevo_House_Hovercraft"));
      PSGame.getParty().addQuestItem(PSGame.getItem(OriginalItem.Vehicle_FlowMover));
    } else {
      await PSMenu.Stext(PSGame.getString("Chest_Search"));
    }
    await PSMenu.endScene();
  }

  public static async ent1(): Promise<void> {
    PSGame.EntStart();
    await PSMenu.Stext(PSGame.getString("Bortevo_People_Man_1"));
    PSGame.EntFinish();
  }

  public static exit(): void {
    PSGame.mapswitchToPlanet(Planet.PALMA, CityHelper.getX(City.BORTEVO), CityHelper.getY(City.BORTEVO));
  }
}
